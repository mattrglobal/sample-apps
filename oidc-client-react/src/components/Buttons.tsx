import * as React from 'react';

interface IButtonsProps {
  login: () => void;
  getUser: () => void;
  logout: () => void;
}

const Buttons: React.SFC<IButtonsProps> = props => {
  return (
    <div className="row">
      <div className="col-md-12 text-center" style={{ marginTop: '30px' }}>
        <button className="btn btn-primary btn-login" style={{ margin: '10px' }} onClick={props.login}>
          Verify
        </button>
        <button className="btn btn-secondary btn-getuser" style={{ margin: '10px' }} onClick={props.getUser}>
          Get User info
        </button>
        <button className="btn btn-dark btn-logout" style={{ margin: '10px' }} onClick={props.logout}>
          Logout
        </button>
      </div>
    </div>
  );
};

export default Buttons;
