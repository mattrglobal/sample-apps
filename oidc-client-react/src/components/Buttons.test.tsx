import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Buttons from './Buttons';

describe(Buttons, () => {
  const mockLogin = jest.fn();
  const mockLogout = jest.fn();
  const mockGetUser = jest.fn();

  const component = shallow(
    <Buttons
      login={mockLogin}
      logout={mockLogout}
      getUser={mockGetUser}
    />
  );

  it('renders and matches our snapshot', () => {
    const buttonComponent = renderer.create(
      <Buttons
        login={mockLogin}
        logout={mockLogout}
        getUser={mockGetUser}
      />
    );
    const tree = buttonComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the image and h1', () => {
    expect(component.find('button')).toHaveLength(5);
  });

  it('try to click on login button', () => {
    component.find('button.btn-login').simulate('click');
    expect(mockLogin).toBeCalled();
  });

  it('try to click on logout button', () => {
    component.find('button.btn-logout').simulate('click');
    expect(mockLogout).toBeCalled();
  });
});
