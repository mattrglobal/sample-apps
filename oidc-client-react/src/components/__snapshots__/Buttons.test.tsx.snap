// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders and matches our snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 text-center"
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <button
      className="btn btn-primary btn-login"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Login
    </button>
    <button
      className="btn btn-secondary btn-getuser"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Get User info
    </button>
    <button
      className="btn btn-warning btn-getapi"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Call API
    </button>
    <button
      className="btn btn-success btn-renewtoken"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Renew Token
    </button>
    <button
      className="btn btn-dark btn-logout"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Logout
    </button>
  </div>
</div>
`;
